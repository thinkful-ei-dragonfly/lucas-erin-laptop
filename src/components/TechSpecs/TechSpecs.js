import React from 'react';
import Features from '../Features/Features';
import './TechSpecs.css'

function TechSpecs(props){
  return (
    <section className="main__form">
      <h3>TECH SPECS AND CUSTOMIZATIONS</h3>
      <Features features={props.features}/>
    </section>
  )
}

export default TechSpecs;
