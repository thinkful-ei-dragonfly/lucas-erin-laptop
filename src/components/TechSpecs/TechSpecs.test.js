import React from 'react';
import { shallow } from 'enzyme';
import { configure } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import toJson from 'enzyme-to-json'
import TechSpecs from './Techspecs';
import STORE from '../../STORE.js'
import App from '../../App'
configure({ adapter: new Adapter() });

describe('TechSpecs', () => {
  const features = Object.keys(STORE)
        .map(key => {
          const options = STORE[key].map((item, index) => {
            const selectedClass = item.name === STORE[key].name ? 'feature__selected' : '';
            const featureClass = 'feature__option ' + selectedClass;
            return <li key={index} className="feature__item">
              <div className={featureClass}

                onClick={e => App.updateFeature(key, item)}>
                  { item.name }
                  ({ new Intl.NumberFormat('en-US', { style: 'currency', currency: 'USD'})
                    .format(item.cost) })
              </div>
            </li>
          });

          return <div className="feature" key={key}>
            <div className="feature__name">{key}</div>
            <ul className="feature__list">
              { options }
            </ul>
          </div>
        });
  it('renders with enzyme', () => {
    const wrapper = shallow(<TechSpecs features={features}/>);
    expect(toJson(wrapper)).toMatchSnapshot();
  })
})
