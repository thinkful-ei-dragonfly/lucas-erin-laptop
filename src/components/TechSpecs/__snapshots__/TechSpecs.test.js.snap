// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TechSpecs renders with enzyme 1`] = `
<section
  className="main__form"
>
  <h3>
    TECH SPECS AND CUSTOMIZATIONS
  </h3>
  <Features
    features={
      Array [
        <div
          className="feature"
        >
          <div
            className="feature__name"
          >
            Processor
          </div>
          <ul
            className="feature__list"
          >
            <li
              className="feature__item"
            >
              <div
                className="feature__option "
                onClick={[Function]}
              >
                17th Generation Intel Core HB (7 Core with donut spare)
                (
                $700.00
                )
              </div>
            </li>
            <li
              className="feature__item"
            >
              <div
                className="feature__option "
                onClick={[Function]}
              >
                Professor X AMD Fire Breather with sidewinder technology
                (
                $1,200.00
                )
              </div>
            </li>
          </ul>
        </div>,
        <div
          className="feature"
        >
          <div
            className="feature__name"
          >
            Operating System
          </div>
          <ul
            className="feature__list"
          >
            <li
              className="feature__item"
            >
              <div
                className="feature__option "
                onClick={[Function]}
              >
                Ubuntu Linux 16.04
                (
                $200.00
                )
              </div>
            </li>
            <li
              className="feature__item"
            >
              <div
                className="feature__option "
                onClick={[Function]}
              >
                Bodhi Linux
                (
                $300.00
                )
              </div>
            </li>
          </ul>
        </div>,
        <div
          className="feature"
        >
          <div
            className="feature__name"
          >
            Video Card
          </div>
          <ul
            className="feature__list"
          >
            <li
              className="feature__item"
            >
              <div
                className="feature__option "
                onClick={[Function]}
              >
                Toyota Corolla 1.5v
                (
                $1,150.98
                )
              </div>
            </li>
            <li
              className="feature__item"
            >
              <div
                className="feature__option "
                onClick={[Function]}
              >
                Mind mild breeze 2000
                (
                $1,345.00
                )
              </div>
            </li>
          </ul>
        </div>,
        <div
          className="feature"
        >
          <div
            className="feature__name"
          >
            Display
          </div>
          <ul
            className="feature__list"
          >
            <li
              className="feature__item"
            >
              <div
                className="feature__option "
                onClick={[Function]}
              >
                15.6" UHD (3840 x 2160) 60Hz Bright Lights and Knobs
                (
                $1,500.00
                )
              </div>
            </li>
            <li
              className="feature__item"
            >
              <div
                className="feature__option "
                onClick={[Function]}
              >
                15.3" HGTV (3840 x 2160) Home makeover edition
                (
                $1,400.00
                )
              </div>
            </li>
          </ul>
        </div>,
      ]
    }
  />
</section>
`;
